### PR TITLE
Add 'compose-docker.yml' in filters.yml

### LINKED ISSUE #2575: Changes to Docker Compose should trigger a full CI + CD run
## Description

We use filters to determine which jobs to run for each PR, this helps keep the CI + CD workflow fast and prevents unnecessary jobs from running. These filters are in the https://github.com/WordPress/openverse/blob/9a1bf60c02a92f63224cd43f2644bda2219e3b63/.github/filters.yml file.

## Problem

Since the `docker-compose.yml` file is not included in any of the filters, when changes are made to it, most jobs are skipped or bypassed. This reduces confidence in the validity of the changes.

## Reproduction

See job run https://github.com/WordPress/openverse/actions/runs/5468016976/jobs/9955141021 on PR #2568.

### PR DESCRIPTION
**Pull request by YashGaur000**
## Fixes

Fixes #2575 by @dhruvkb 

## Description

The docker-compose.yml file needed to be added in the .github/filters.yml file under the keys catalog, api, ingestion_server and analytics keys. This would indicate to CI + CD workflow that any change to the Docker Compose file means that these stacks have changed and their tests need to be executed. Now that I added these files it is working fine.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible
      errors.
- [ ] I ran the DAG documentation generator (only applicable for catalog PRs).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by YashGaur000** [comment]
I have edited it according to pull request template. Please review it ASAP.

**Comment by YashGaur000** [comment]
I have edited it according to pull request template. If there's any
other problems, please let me know I fix/edit it quickly.

File: documentation/frontend/guides/quickstart.md [hunk 1]
```diff
@@ -39,10 +39,11 @@ you need to run this.
    ```
 
 4. To bring up the frontend, we have another `just` recipe. We have `just`
-   recipes for almost everything.
+   recipes for almost everything. Also, build it with the second command.
```
Comment (AetherUnbound) [inline documentation/frontend/guides/quickstart.md:42]: These lines look out of place and don't apply to the change mentioned in this PR. Do you mind removing them?

**Review by AetherUnbound — CHANGES_REQUESTED** [review]
The change to `filters.yml` looks good! Do you mind removing the changes to `quickstart.md`? They seem unrelated to this issue. Thanks!

File: documentation/frontend/guides/quickstart.md (hunk 1 above)
Comment (YashGaur000) [reply documentation/frontend/guides/quickstart.md]: Done mam. If there's anything else please tell me.

File: documentation/frontend/guides/quickstart.md [hunk 2]
```diff
@@ -43,6 +43,7 @@ you need to run this.
 
    ```console
    $ just frontend/run dev
+   $ pnpm --filter='@openverse/eslint-plugin' build
```
Comment (obulat) [inline documentation/frontend/guides/quickstart.md:46]:
```suggestion
```

**Review by obulat — CHANGES_REQUESTED** [review]
I've added the code suggestions to remove the unrelated changes. After that, this PR should be ready to go.

In the future, @YashGaur000, do not open PRs from the `main` branch of your fork: it will be difficult to create another PR, or update your PR.

**Comment by YashGaur000** [comment]
Done mam. Please review it.

**Comment by AetherUnbound** [comment]
@YashGaur000 If you look at https://github.com/WordPress/openverse/pull/2652/files, you can see there are several changes unrelated to the ones necessary for this PR in `.github/filters.yml`. Please remove those changes from the pull requests so that we only have the changes present in `.github/filters.yml`. It may be easier to close this pull request and open a new one on a separate branch of your fork, to get around the issue Olga had mentioned.

**Comment by YashGaur000** [comment]
Done mam.

**Comment by YashGaur000** [comment]
I did stuff you recommend. Please check if still not align with your recommended steps then please tell me more to resolve this issue so that I don't make this mistake again in future contribution.

PLEASE PLEASE HELP 😢😭

**Comment by AetherUnbound** [comment]
I've responded in your new PR, #2687

**Comment by AetherUnbound** [comment]
Closing in favor of #2687

**Pull request closed without merging**